src/permutat.cc

```diff
@@ -1346,27 +1346,21 @@ static inline Obj CYCLE_STRUCT_PERM(Obj perm)
     ende = cnt;
 
     /* create the list */
-    list = NEW_PLIST(T_PLIST, max - 1);
-    SET_LEN_PLIST(list, max - 1);
+    list = NEW_PLIST(T_PLIST, max);
+    SET_LEN_PLIST(list, max);
     ptList = ADDR_OBJ(list);
 
     /* Recalculate after possible GC */
     scratch = ADDR_TMP_PERM<T>();
     offset = (T *)((UInt)scratch + (bytes));
 
-    for (pnt = 1; pnt < max; pnt++) {
-        ptList[pnt] = 0;
-    } /* clean out */
-
     for (cnt = 0; cnt < ende; cnt++) {
         pnt = (UInt)offset[cnt];
-        pnt--;
-        ptList[pnt] = (Obj)((UInt)ptList[pnt] + 1);
-    }
-
-    for (pnt = 1; pnt < max; pnt++) {
-        if (ptList[pnt] != 0) {
-            ptList[pnt] = INTOBJ_INT((UInt)ptList[pnt]);
+        if(ptList[pnt] == 0) {
```

Comment:
```suggestion
        if (ptList[pnt] == 0) {
```

Reply:
One day I'll remember to run clang-format